omod-2.0/src/test/java/org/openmrs/module/attachments/rest/AttachmentRestController2_0Test.java

```diff
@@ -94,6 +99,58 @@ public void shouldGetRefByUuid() {
 	public void shouldGetFullByUuid() {
 	}
 	
+	@Test
+	public void createAttachment_shouldCreateANewAttachment() throws Exception {
+		SimpleObject attachment = new SimpleObject();
+		attachment.add("uuid", obs.getUuid());
+		attachment.add("comment", obs.getComment());
+		attachment.add("complexData", obs.getComplexData());
+		
+		String json = new ObjectMapper().writeValueAsString(attachment);
+		
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+		req.setContent(json.getBytes());
+		
+		Util.log("Created attachment", attachment);
+		
+		// Check existence in database
+		String uuid = (String) attachment.get("uuid");
+		Assert.assertNotNull("uuid", obs.getUuid());
+		Assert.assertNotNull("comment", obs.getComment());
+		Assert.assertEquals("complexData", obs.getComplexData(), null);
```

Comment:
@sherrif10 , you instead need to check the deserialised response , not the very json object you posted